online-shop/src/app/auth/login/login.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Route, Router } from '@angular/router';
+import { AuthService } from '../auth.service';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+})
+export class LoginComponent {
+  formGroup: FormGroup;
+
+  constructor(
+    private fb: FormBuilder,
+    private authService: AuthService,
+    private router: Router
+  ) {
+    this.formGroup = this.fb.group({
+      username: ['', [Validators.required, Validators.maxLength(20)]],
+      password: ['', [Validators.required, Validators.maxLength(20)]],
+    });
+  }
+
+  onSubmit() {
+    let user = {
+      username: this.formGroup.get('username')?.value,
+      password: this.formGroup.get('password')?.value,
+    };
+    let request = this.authService.logIn(user);
+    request.subscribe(
```

Comment:
this method as you worte it seems to be deprecated, please check the docs and substitute it with one that isn't